libqtile/dgroups.py

```diff
@@ -127,8 +127,13 @@ def _setup_groups(self):
             self.add_dgroup(group, group.init)
 
             if group.spawn and not self.qtile.no_spawn:
-                pid = self.qtile.cmd_spawn(group.spawn)
-                self.add_rule(Rule(Match(net_wm_pid=[pid]), group.name))
+                if isinstance(group.spawn, str):
```

Comment:
You may want to use `six.string_types` here instead of `str` to be compatible with py2/3.